components/advanced_create_post/advanced_create_post.test.jsx

```diff
@@ -1239,6 +1269,36 @@ describe('components/advanced_create_post', () => {
         expect(wrapper.state('message')).toBe(markdownTable);
     });
 
+    it('should be able to format a pasted hyperlink', () => {
+        const wrapper = shallow(advancedCreatePost());
+        const mockImpl = () => {
+            return {
+                setSelectionRange: jest.fn(),
+                focus: jest.fn(),
+            };
+        };
+        wrapper.instance().textboxRef.current = {getInputBox: jest.fn(mockImpl), focus: jest.fn(), blur: jest.fn()};
+
+        const event = {
+            target: {
+                id: 'post_textbox',
+            },
+            preventDefault: jest.fn(),
+            clipboardData: {
+                items: [1],
+                types: ['text/html'],
+                getData: () => {
+                    return '<a href="https://test.domain">link text</a>';
+                },
+            },
+        };
+
+        const markdownLink = '[link text](https://test.domain)';
+
+        wrapper.instance().pasteHandler(event);
+        expect(wrapper.state('message')).toBe(markdownLink);
```

Comment:
could you add a check for the change in a draft here as well please. Not really needed, but better to be more careful about these to catch more errors

Reply:
as far as I see I think the pasteHandler in `advancedCreatePost` doesn't actually update the draft, only does it in comment's pasteHandler

Reply:
yes, it does it [here](https://github.com/mattermost/mattermost-webapp/blob/5f378028908969463c57af109755b2b7de30cc5f/components/advanced_create_post/advanced_create_post.tsx#L362)

Reply:
Took a further look into testing, tried to set something up but draft didn't seem to update, did some actual manual testing and looks like it doesn't save draft when pasting tables/links. (both before this change (when pasting tables) and now (when any paste goes through the handler)). Should be a relatively simple addition to fix but should this probably be another ticket? @esethna